### DISCUSSION TITLE
Unexpected exception: Cannot bind argument to parameter 'Credential' because it is null. at 419

### DISCUSSION THREAD

**Discussion by gw-terry** [tomasrudh/AutoShutdownSchedule#6] [Q&A]
Hi,

I'm pretty new to Azure automation so it's likely I'm doing something wrong - but I keep receiving the below error when trying to run the script. Any ideas?

```
Failed
Unexpected exception: Cannot bind argument to parameter 'Credential' because it is null. at 419 (Unexpected exception: Cannot bind argument to parameter 'Credential' because it is null. at 419)
```

**Comment by tomasrudh** [comment]
Hi,

What type of credential are you using? The easiest to set setup is a RunAs account or a System Managed Account. Microsoft is recommending to move away from RunAs accounts, as they require some work regularly, renewing the certificate.

I would suggest you setup a System Managed Account, and follow the instructions in the documentation, if you have not already done that.

Regards,
Tomas

**Comment by tomasrudh** [comment]
I have found that in some circumstances did the credentials not work. I have published a new version that will fix this.
